nuxeo-core/nuxeo-core-bulk/src/main/java/org/nuxeo/ecm/core/bulk/action/computation/SortBlob.java

```diff
@@ -50,6 +55,32 @@ public void processRecord(ComputationContext context, String inputStreamName, Re
         String commandId = in.getCommandId();
         Blob blob = getBlob(in.getDataAsString());
         blob = sort(blob, commandId);
+
+        File blobFile = blob.getFile();
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(blobFile));
+            // Add header
+            String header = in.getHeaderAsString();
+            StringBuilder result = StringUtils.isNotEmpty(header) ? new StringBuilder(String.format("%s%n", header))
+                    : new StringBuilder();
```

Comment:
just create your stringbuilder and append the header if not empty like for the footer